### PR TITLE
scicomp/ssh: Updates throughout the SSH page

### PR DESCRIPTION
**Pull request by rkdarst**
- Generalize the page, since SSH config and keys are going to become
  much more important in the future.
- Better explain all the key options for connecting on different
  operating systems.

### CODE REVIEW

**Comment by rkdarst** [comment]
Ready for review, please review quickly

**Comment by Gehock** [comment]
I've seen some recommendations about using ed25519 keys instead of RSA (for example, [github docs](https://docs.github.com/en/authentication/connecting-to-github-with-ssh/generating-a-new-ssh-key-and-adding-it-to-the-ssh-agent)), although couldn't find any official guidance. If nothing else, ed25519 keys are nicer to handle because they're shorter (in terms of characters) than RSA. There's also a [list](https://www.aalto.fi/en/services/linux-shell-servers-at-aalto) or fingerprints of Aalto shell servers available.

File: scicomp/ssh.rst [hunk 1]
```diff
@@ -260,55 +392,53 @@ and you would use ``ssh triton`` to log in. Any additional server configs can fo
...
 
 
 .. _example_config_for_ssh:
 
 Full sample config file
 ~~~~~~~~~~~~~~~~~~~~~~~
 
-The following code is placed in the config file created above (i.e. ``~/.ssh/config`` on mac/linux or ``%USERPROFILE%.ssh/config`` on windows) ::
+The following code is placed in the config file created above
+(i.e. ``~/.ssh/config`` on Mac/Linux or ``%USERPROFILE%.ssh/config``
```
Comment (Gehock) [inline scicomp/ssh.rst:415]:
```suggestion
(i.e. ``~/.ssh/config`` on Mac/Linux or ``%USERPROFILE%/.ssh/config``
```
Fixing a typo

**Comment by rkdarst** [comment]
> I've seen some recommendations about using ed25519 keys instead of RSA (for example, [github docs](https://docs.github.com/en/authentication/connecting-to-github-with-ssh/generating-a-new-ssh-key-and-adding-it-to-the-ssh-agent)), although couldn't find any official guidance. If nothing else, ed25519 keys are nicer to handle because they're shorter (in terms of characters) than RSA.

I thought of that, but wondered, could it be there are still some common machines around that don't support ec25519.  At least Triton does so that's good.  With the decommissioning of taltta maybe everything does?

**Pull request merged**